File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>
#define MAX 100000
using namespace std;
int arr[MAX+1];

int dp(int n) {
    arr[0] = 0;
    arr[1] = 1;
    for (int i = 2; i <= n; i++) {
        arr[i] = (arr[i-1]+arr[i-2])%1234567;
    }
    return arr[n];
}

int solution(int n) {
    int answer = dp(n);
    return answer;
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    answer = 0
    arr = [0, 1]
    arr[0] = 0
    arr[1] = 1
    for i in range(2, n+1):
        arr.append(arr[i-1] + arr[i-2])
    answer = arr[n] % 1234567
    return answer